main.go

```diff
@@ -282,3 +289,17 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func getAvailableCRDNames(ctx context.Context, cl apiclient.Client) (map[string]bool, error) {
+	crdExist := map[string]bool{}
+	crdList := &metav1.PartialObjectMetadataList{}
+	crdList.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinitionList"))
+	if err := cl.List(ctx, crdList); err != nil {
+		return nil, fmt.Errorf("error getting CRD, %v", err)
+	}
+	// Iterate over the list and populate the map
+	for i := 0; i < len(crdList.Items); i++ {
```

Comment:
Please use the range variant when listing full slices
```suggestion
	for i  := range crdList.Items {
```